src/axelar/AxelarTranceiver.sol

```diff
@@ -0,0 +1,130 @@
+    error NotImplemented();
+
+    constructor(address _gateway, address _gasService, address _manager)
+        AxelarExecutable(_gateway)
+        Managed(_manager)
+    {
+        gasService = IAxelarGasService(_gasService);
+    }
+
+    function _setup(bytes calldata params) internal override {
+
+    }
+
+    function contractId() external pure override returns (bytes32) {
+        return keccak256('axelar-transceiver');
+    }
+
+    /**
+     * Set the bridge manager contract address
+     * @param chainId The chainId of the chain. This is used to identify the chain in the EndpointManager.
+     * @param chainName The chainName of the chain. This is used to identify the chain in the AxelarGateway.
+     */
+    function setAxelarChainId(uint16 chainId, string calldata chainName, string calldata axelarAddress) external onlyOwner {
+        idToAxelarChainIds[chainId] = chainName;
+        axelarChainIdToId[chainName] = chainId;
+        idToAxelarAddress[chainId] = axelarAddress;
+        axelarAddressToId[axelarAddress] = chainId;
+    }
+
+    /// @notice Fetch the delivery price for a given recipient chain transfer.
+    /// param recipientChain The Wormhole chain ID of the target chain.
+    /// param instruction An additional Instruction provided by the Transceiver to be
+    ///        executed on the recipient chain.
+    /// @return deliveryPrice The cost of delivering a message to the recipient chain,
+    ///         in this chain's native token.
+    function quoteDeliveryPrice(
+        uint16 /*recipientChain*/,
+        TransceiverStructs.TransceiverInstruction memory /*instruction*/
+    ) external view override virtual returns (uint256) {
+        // Axelar doesn't support on-chain gas fee.
+        return 0;
+    }
+
+    /// @dev Send a message to another chain.
+    /// @param recipientChainId The Wormhole chain ID of the recipient.
+    /// param instruction An additional Instruction provided by the Transceiver to be
+    /// executed on the recipient chain.
+    /// @param nttManagerMessage A message to be sent to the nttManager on the recipient chain.
+    function sendMessage(
+        uint16 recipientChainId,
+        TransceiverStructs.TransceiverInstruction memory /*instruction*/,
+        bytes memory nttManagerMessage,
+        bytes32 recipientNttManagerAddress,
+        bytes32 refundAddress
+    ) external payable override virtual onlyManager() {
+        emit SendTransceiverMessage(recipientChainId, nttManagerMessage, recipientNttManagerAddress, refundAddress);
+
+        string memory destinationContract = idToAxelarAddress[recipientChainId];
+        string memory destinationChain = idToAxelarChainIds[recipientChainId];
+
+        if(bytes(destinationChain).length == 0 || bytes(destinationContract).length == 0) revert InvalidChainId(recipientChainId);
+
+        bytes memory payload = abi.encode(manager, nttManagerMessage, recipientNttManagerAddress);
```

Comment:
length prefixing the messages is a good idea. allows for more efficient deserialization.

Reply:
whiche messages? There is a single message named `nttManagerMessage` i thought, am I wrong here?

Reply:
yes`nttManagerMessage`. See https://github.com/wormhole-foundation/example-native-token-transfers/blob/0e861be07c0b915b9bdf1040ef78c443b6db70b0/evm/src/libraries/TransceiverStructs.sol#L225 and https://github.com/wormhole-foundation/example-native-token-transfers/blob/0e861be07c0b915b9bdf1040ef78c443b6db70b0/evm/src/libraries/TransceiverStructs.sol#L199-L209 as an example.

Reply:
fixed